editor/src/components/navigator/navigator-utils.ts

```diff
@@ -793,7 +805,57 @@ function getNavigatorRowsForTree(
   return condensedTree.flatMap((t) => walkTree(t, 0))
 }
 
-export function getNavigatorTargets(
+export function getNavigatorTargetsFromEditorState(
+  editorState: EditorState,
+): GetNavigatorTargetsResults {
+  return navigatorTargetsSelector({ editor: editorState })
+}
+
+export function useGetNavigatorTargets(): GetNavigatorTargetsResults {
+  return useEditorState(
+    // TODO try not to use fullStore here
+    Substores.fullStore,
+    navigatorTargetsSelector,
+    'useGetNavigatorTargets',
+  )
+}
+
+export const navigatorTargetsSelector = createSelector(
```

Comment:
the most important part of the PR is this new selector, and the helper functions using it